### PR TITLE
Optimize plugin startup performance

### PR DESCRIPTION
**Pull request by van800**
Avoid using JsonConvert.SerializeObject which is slow, use StringBuilder instead.

### CODE REVIEW

File: unity/EditorPlugin/NonUnity/RiderProtocolController.cs [hunk 1]
```diff
@@ -46,7 +48,15 @@ public ProtocolInstance(int port, string solutionName)
...
+        //return JsonConvert.SerializeObject(connections); //turns out to be slow https://github.com/JetBrains/resharper-unity/issues/728 
+      var sb = new StringBuilder("[");
+
+      sb.Append(connections
+        .Select(connection=> "{" + $"\"Port\":{connection.Port},\"SolutionName\":\"{connection.SolutionName}\"" + "}")
+        .Aggregate((a, b) => a + "," + b));
+
+      sb.Append("]");
+      return sb.ToString();
     }
```
Comment (citizenmatt) [inline unity/EditorPlugin/NonUnity/RiderProtocolController.cs]: A `foreach` might be even better here - LINQ can be quite expensive.

**Comment by citizenmatt** [comment]
Does this mean we can get rid of the JSON.NET reference? That will make our plugin smaller, which should make it load (a tiny bit) faster.

File: unity/EditorPlugin/NonUnity/RiderProtocolController.cs (hunk 1 above)
Comment (van800) [reply unity/EditorPlugin/NonUnity/RiderProtocolController.cs]: I don't like removing trailing comma, Aggregate is safer. I can hardly imagine someone having more than 3-4 solution files in this folder.

**Comment by van800** [comment]
There is one more usage https://github.com/JetBrains/resharper-unity/blob/182/unity/EditorPlugin/RiderPathLocator.cs#L290

**Comment by citizenmatt** [comment]
Ah. What defines do we normally have set? Both projects in the solution have `UNITY_2017_3` set - do we ever build against an earlier version of UnityEngine.dll?

**Comment by van800** [comment]
It should be a problem. I will check some old Unity version.
We do build in `unity\JetBrains.Rider.Unity.Editor.sln`, but it is for validation purposes only.

**Comment by van800** [comment]
Yeah, it brakes Unity 4.7 and 5.0
Could not load type 'UnityEngine.JsonUtility' from assembly 'JetBrains.Rider.Unity.Editor.Plugin.Repacked'.
  at JetBrains.Rider.Unity.Editor.RiderPathLocator+<>c__DisplayClass13_0.<CollectPathsFromToolbox>b__1 (System.String a) [0x00000] in <filename unknown>:0

**Comment by van800** [comment]
Added fix https://github.com/JetBrains/resharper-unity/pull/762

**Pull request merged**